neo/Consensus/ConsensusContext.cs

```diff
@@ -405,6 +423,38 @@ public void Reset(byte viewNumber)
             TransactionHashes = null;
             PreparationPayloads = new ConsensusPayload[Validators.Length];
             if (MyIndex >= 0) LastSeenMessage[MyIndex] = (int)Block.Index;
+
+            if (resetFutures)
+                ResetFuturePayloads();
+        }
+
+        public void ResetFuturePayloads()
+        {
+            FutureCommitPayloads = new ConsensusPayload[Validators.Length];
+            FuturePreparationPayloads = new ConsensusPayload[Validators.Length];
+            FutureChangeViewPayloads = new ConsensusPayload[Validators.Length];
+            FutureRecoveryPayloads = new ConsensusPayload[Validators.Length];
+        }
+
+        public bool HasFuturePayloads()
+        {
+            if (FuturePreparationPayloads != null)
+                foreach (var payload in FuturePreparationPayloads)
+                    if (payload != null)
+                        return true;
```

Comment:
Hi @vncoelho, is there any reason this has to written this way?  Do we need this foreach?

Reply:
No, it can be in another format.

HasFuturePayloads is used on InitializeConsensus.

Reply:
Improved using a Counter as suggested by @shargon